### PR TITLE
fix opening first mailbox.

### LINKED ISSUE #659: Automatically choose the first inbox when opening the app
### Steps to reproduce

1. "Login" to your mail (Open your mail in ownCloud)

### Actual behaviour

The mail is loading, and keeps loading until you press any folder in the mailbox.

http://i.imgur.com/y5FdUwV.png

### Expected behaviour

The first folder in the mailbox should be opened automatically

### Server and Version

8.0.3
PHP 5.5
Mail 0.1

### PR DESCRIPTION
**Pull request by irgendwie**
should fix #659 (very ugly, but functional :see_no_evil:)
I debugged a few hours and it seems like there is a race condition,
if interested I can explain at the next meetup in munich.
I'm too lazy to write a full description of the problem here.

@enoch85 @wurstchristoph please review!
also cc @jancborchardt @DeepDiver1975

### CODE REVIEW

**Comment by irgendwie** [comment]
Short description of the problem:
It happens, that the folders of the first mail account get loaded first and then it tries to load the messages and set the `active` flag to false.
As the second and other mail accounts didn't load the folders by now, the `localAccount` variable is undefined.

**Comment by enoch85** [comment]
@irgendwie could you please fix the failed tests and I will review this later.

**Comment by jancborchardt** [comment]
@enoch85 review again. :) (@irgendwie if you changed something you manually need to let folks know – we don’t get notifications about new commits/pushes)

**Comment by enoch85** [comment]
Doesn't fix the error for me. Still have to click the first inbox manually. :-1:

**Comment by irgendwie** [comment]
Did you try it in a private browsing session? At least the error should be gone!

**Comment by enoch85** [comment]
@irgendwie Nope, not a private session. 

Tested in IE and it works, but not in Chrome. Haven't Firefox installed so I didn't try that.

**Comment by irgendwie** [comment]
@enoch85 it works for me in Chromium, maybe you still have a cached version of the *.js file or it doesn't work for you.

**Comment by enoch85** [comment]
@irgendwie Yes, sorry. My bad. Tested Chrome in my wifes PC and it works. :+1: 

@jancborchardt Ok to merge

**Comment by enoch85** [comment]
Btw, thank you @irgendwie for this fix!

**Comment by jancborchardt** [comment]
We need a second review to be able to merge. @wurstchristoph can you also review since you’re the other person who can reproduce it?

**Comment by wurstchristoph** [comment]
@jancborchardt sure, I'll test it tomorrow ;-)

File: js/mail.js [hunk 1]
```diff
@@ -709,6 +709,9 @@ var Mail = {
 			// disable all other folders for all accounts
 			_.each(Mail.State.accounts, function (account) {
 				var localAccount = Mail.State.folderView.collection.get(account.accountId);
+				if (localAccount === undefined) {
```
Comment (DeepDiver1975) [inline js/mail.js]: please use

```js
_.isUndefined(localAccount)

**Comment by irgendwie** [comment]
@DeepDiver1975 changed the function accordingly.

**Comment by wurstchristoph** [comment]
@irgendwie @jancborchardt fixed the problem for me :+1:

**Comment by jancborchardt** [comment]
Merging it in then. :)

**Pull request merged**